### PR TITLE
check-pause-resume: bugfix in expect script

### PR DESCRIPTION
**Pull request by Bin-QA**
fix escape '/' for repeat_count in dynamic create expect script

fix issue #49 

Signed-off-by: Wu, BinX <binx.wu@intel.com>

### CODE REVIEW

**Comment by xiulipan** [comment]
@Bin-QA Also could you make the delay range for the pause/resume configurable. So we can test fast pause/resume and very slow pause/resume.

File: test-case/check-pause-resume.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
 OPT_OPT_lst['t']='tplg'     OPT_DESC_lst['t']='tplg file, default value is env TPLG: $TPLG'
 OPT_PARM_lst['t']=1         OPT_VALUE_lst['t']="$TPLG"
 
-OPT_OPT_lst['m']='mode'    OPT_DESC_lst['m']='test mode'
+OPT_OPT_lst['m']='mode'     OPT_DESC_lst['m']='test mode'
```
Comment (xiulipan) [inline test-case/check-pause-resume.sh]: seems to have some wrong space added?

File: test-case/check-pause-resume.sh [hunk 2]
```diff
@@ -39,6 +45,10 @@ test_mode=${OPT_VALUE_lst['m']}
 repeat_count=${OPT_VALUE_lst['c']}
 #TODO: file name salt for capture
 file_name=${OPT_VALUE_lst['f']}
+# configure random value range
+rnd_min=${OPT_VALUE_lst['i']}
+rnd_max=${OPT_VALUE_lst['a']}
+rnd_range=$[ $rnd_max - $rnd_min ]
```
Comment (xiulipan) [inline test-case/check-pause-resume.sh]: Do we need more check here to make sure user input valid value?
EG: check if max >= mix

File: test-case/check-pause-resume.sh (hunk 1 above)
Comment (Bin-QA) [reply test-case/check-pause-resume.sh]: alignment with other line

File: test-case/check-pause-resume.sh (hunk 2 above)
Comment (Bin-QA) [reply test-case/check-pause-resume.sh]: add basic logic is ok for me,  
but for the other hand,  
whether need to the value is string/number for the all parameter

Comment (xiulipan) [reply test-case/check-pause-resume.sh]: Then let add a TODO here.

**Comment by libinyang** [comment]
LGTM

File: test-case/check-pause-resume.sh [hunk 3]
```diff
@@ -29,6 +29,12 @@ OPT_PARM_lst['c']=1         OPT_VALUE_lst['c']=10
 OPT_OPT_lst['f']='file'     OPT_DESC_lst['f']='file name'
 OPT_PARM_lst['f']=1         OPT_VALUE_lst['f']=''
 
+OPT_OPT_lst['i']='min'      OPT_DESC_lst['i']='random rang min value, unit is ms'
```
Comment (aiChaoSONG) [inline test-case/check-pause-resume.sh]: what is "rang" mean here, didn't get it here? you mean "run"?

File: test-case/check-pause-resume.sh [hunk 4]
```diff
@@ -29,6 +29,12 @@ OPT_PARM_lst['c']=1         OPT_VALUE_lst['c']=10
...
+OPT_PARM_lst['i']=1         OPT_VALUE_lst['i']='100'
+
+OPT_OPT_lst['a']='max'      OPT_DESC_lst['a']='random rang max value, unit is ms'
```
Comment (aiChaoSONG) [inline test-case/check-pause-resume.sh]: same as above, "rang"

File: test-case/check-pause-resume.sh [hunk 5]
```diff
@@ -39,6 +45,11 @@ test_mode=${OPT_VALUE_lst['m']}
 repeat_count=${OPT_VALUE_lst['c']}
 #TODO: file name salt for capture
 file_name=${OPT_VALUE_lst['f']}
+# configure random value range
+rnd_min=${OPT_VALUE_lst['i']}
+rnd_max=${OPT_VALUE_lst['a']}
+rnd_range=$[ $rnd_max - $rnd_min ]
+[[ $rnd_range -le 0 ]] && dlogw "Error random rang scope [ min:$rnd_min - max:$rnd_max ]" && exit 2
```
Comment (aiChaoSONG) [inline test-case/check-pause-resume.sh]: "rang" here too

File: test-case/check-pause-resume.sh [hunk 6]
```diff
@@ -79,15 +90,15 @@ spawn $cmd -D $dev -r $rate -c $channel -f $fmt -vv -i $file_name -q
 set i 1
 expect {
     "*#*+*\%" {
-        set sleep_t [expr int([expr rand() * 1000]) + 1000 ]
-        puts "\r(\$i/\$repeat_count) Wait for \$sleep_t ms for pause"
+        set sleep_t [expr int([expr rand() * $rnd_range]) + $rnd_min ]
+        puts "\r(\$i/$repeat_count) Wait for \$sleep_t ms for pause"
```
Comment (aiChaoSONG) [inline test-case/check-pause-resume.sh]: change "Wait for \$sleep_t ms for pause"  to "Wait for \$sleep_t ms before pause"

File: test-case/check-pause-resume.sh [hunk 7]
```diff
@@ -79,15 +90,15 @@ spawn $cmd -D $dev -r $rate -c $channel -f $fmt -vv -i $file_name -q
...
         send " "
         after \$sleep_t
         exp_continue
     }
     "*PAUSE*" {
-        set sleep_t [expr int([expr rand() * 1000]) + 1000 ]
-        puts "\r(\$i/\$repeat_count) Wait for \$sleep_t ms for resume"
+        set sleep_t [expr int([expr rand() * $rnd_range]) + $rnd_min ]
+        puts "\r(\$i/$repeat_count) Wait for \$sleep_t ms for resume"
```
Comment (aiChaoSONG) [inline test-case/check-pause-resume.sh]: change "Wait for \$sleep_t ms for resume" to "Wait \$sleep_t ms before resume"

**Comment by Bin-QA** [comment]
@aiChaoSONG fix typo

**Comment by Bin-QA** [comment]
@aiChaoSONG please help to review

**Pull request merged**